esp/smc/SMCLib/TpContainer.cpp

```diff
@@ -0,0 +1,781 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma warning (disable : 4786)
+// TpWrapper.cpp: implementation of the CTpWrapper class.
+//
+//////////////////////////////////////////////////////////////////////
+
+#include "TpWrapper.hpp"
+#include <stdio.h>
+#include "workunit.hpp"
+#include "exception_util.hpp"
+#include "portlist.h"
+#include "daqueue.hpp"
+#include "dautils.hpp"
+#include "dameta.hpp"
+
+const char* MSG_FAILED_GET_ENVIRONMENT_INFO = "Failed to get environment information.";
+
+//////////////////////////////////////////////////////////////////////
+// Construction/Destruction
+//////////////////////////////////////////////////////////////////////
+
+void CTpWrapper::getClusterMachineList(double clientVersion,
+                                       const char* ClusterType,
+                                       const char* ClusterPath,
+                                       const char* ClusterDirectory,
+                                       IArrayOf<IEspTpMachine> &MachineList,
+                                       bool& hasThorSpareProcess,
+                                       const char* ClusterName)
+{
+    IWARNLOG("UNIMPLEMENTED: CONTAINERIZED(CTpWrapper::getClusterMachineList)");
```

Comment:
Because the code without this change would not have thrown an exception, but would have reported an error.  I tried to make the changes so the behaviour would not change (and break functionality that half works at the moment).